Arm Manual Control

Hydro111 opened the pull request:
The compare has been tested and can successfully move the arm with manual control. The PID Tuning isn't ideal, but it works.

JasperDavidson committed:
Added functionality for Reid's Grabber

Will probably require testing

JasperDavidson committed:
Added functionality for Reid's Grabber

Will probably require testing

JasperDavidson committed:
Added functionality for Reid's Grabber

Will probably require testing

JasperDavidson committed:
Changed a few things

Very descriptive

Hydro111 committed:
Add arm PIDs and inverse kinematics

Hydro111 committed:
Add arm PIDs and inverse kinematics

critter-ACentauri committed:
rebased

critter-ACentauri committed:
resolved conflicts

Blockinator1 committed:
Put some Wrist stuff in - Hale, need to talk to you about the math stuff

Blockinator1 committed:
Put some Wrist stuff in - Hale, need to talk to you about the math stuff

Blockinator1 committed:
Finished Wrist stuff in subsystem

Blockinator1 committed:
Finished Wrist stuff in subsystem

Hydro111 committed:
Rebase Arm-Wrist

Hydro111 committed:
Finish rebase changes

Hydro111 committed:
Merge branch 'Wrist-&-Arm' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into Wrist-&-Arm

Hydro111 committed:
Finalize rebase and merge changes

Remove anomalous TODOs

Mothandman committed:
./

Blockinator1 committed:
Added moveByXYThetaRollOffset (finished LegAnkleSubsystem)

Mothandman committed:
Merge branch 'Wrist-&-Arm' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into Wrist-&-Arm

Mothandman committed:
Add WristCommand

Mothandman committed:
Made Wrist Command

Hydro111 committed:
Fix some weirdness caused by fixing git problems

Hydro111 committed:
Add LegAnkleSubsystem moveByXYTheta speed multipliers

Mothandman committed:
Add joystick control to WristCommand

Hydro111 committed:
Update LegAnkleSubsystem comments

Hydro111 committed:
Merge branch 'dev' into Grabber

Hydro111 committed:
Merge pull request #11 from amesrobotics/Grabber

Grabber to Dev

Hydro111 committed:
Merge pull request #10 from amesrobotics/Wrist-&-Arm

Wrist & arm to Dev

Hydro111 committed:
Reserve PlaceGamePiece command group files

Create MoveArmToTarget command

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

Hydro111 committed:
Bind x to place piece

Finish PlaceGamePiece command setup

Hydro111 committed:
Clean RobotContainer

Hydro111 committed:
Begin ReleaseGameObject work

NotAYoutuberYT committed:
add minor tweaks

Hydro111 committed:
Merge branch 'dev' into swerve-testing

Hydro111 committed:
Merge pull request #14 from amesrobotics/swerve-testing

Merge swerve stuff to dev
Hopefully the solving merge conflicts I did won't cause any problems

Hydro111 committed:
Add drive train PlaceGamePiece functionality

Hydro111 committed:
Make Arm's moveToXYTheta return whether it's done

Hydro111 committed:
Partially complete MoveRobotToGrid

Hydro111 committed:
Input LegAnkleSubsystem constants

Make LegAnkleSubsystem moveToXYTheta safer

Hydro111 committed:
Add indexes to MoveRobotToGrid

Hydro111 committed:
Add current limits for LegAnkleSubsystem

NotAYoutuberYT committed:
make swerve auto very accurate

NotAYoutuberYT committed:
Merge branch 'swerve-testing' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into swerve-testing

Hydro111 committed:
Initialize ShuffleboardSubsystem

NotAYoutuberYT committed:
take advantage of command not being used

CrystalCanStay committed:
Added slow mode multiplier :>

NotAYoutuberYT committed:
allow lazy trajecory following

NotAYoutuberYT committed:
Merge branch 'swerve-testing' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into swerve-testing

NotAYoutuberYT committed:
Merge branch 'swerve-testing' into dev

critter-ACentauri committed:
created FieldPosManager, changed Constants and idk what i did in robotcontainer don't ask

critter-ACentauri committed:
merged FieldPosManager into DEV

NotAYoutuberYT committed:
the robot runs smile emoji

NotAYoutuberYT committed:
do a little tuning of pids

NotAYoutuberYT committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

May0san committed:
Update FieldPosManager.java

CrystalCanStay committed:
Ported over Mason's code. Currently just example.

CrystalCanStay committed:
Fixed previous errors. Now ready to be integrated

NotAYoutuberYT committed:
remove .idea folder (sorry everyone using vscode)

May0san committed:
Update FieldPosManager.java

May0san committed:
Update FieldPosManager.java

May0san committed:
wrote skeleton code for getting poses on the field

should be ready to plug constants in

May0san committed:
Update FieldPosManager.java

May0san committed:
Update FieldPosManager.java

Mothandman committed:
Added constants for the arm min and max

Mothandman committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

Hydro111 committed:
Fix LegAnkleSubsystem PIDs and logic

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

Hydro111 committed:
Modify PlaceGamePiece for integration with FieldPositionManager

May0san committed:
wow added constants and also made a new function yayay

May0san committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

CrystalCanStay committed:
Bug fixing two camera photonvision not complete

May0san committed:
Changed how FieldPosManager uses constants

hopefully this makes entering the constants much easier.

May0san committed:
Update FieldPosManager.java

NotAYoutuberYT committed:
migrate swerve tracking to imu data and integrate FieldPosManager

NotAYoutuberYT committed:
make field position manager in RobotContainer

NotAYoutuberYT committed:
don't have starting position offset in DriveSubsystem

Mothandman committed:
hjfguhkdgnh

Mothandman committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

CrystalCanStay committed:
Photonvision now fully works Version: Snowblind

CrystalCanStay committed:
Merged branch 'PhotonVision' into dev

May0san committed:
Update Constants.java

JasperDavidson committed:
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

Hydro111 committed:
Perform arm extension testing

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

# Conflicts:
#	src/main/java/frc/robot/RobotContainer.java

Mothandman committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

Hydro111 committed:
Test arm pivot

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

CrystalCanStay committed:
Integrated PhotonVision into fieldposmanager

CrystalCanStay committed:
Integrated PhotonVision into fieldPosManager

NotAYoutuberYT committed:
create getMotorsOkTemperature

Mothandman committed:
put shuffleboard widget things in constants. (I'm sorry Hale I know this is an awful commit msg lmao)

Mothandman committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

NotAYoutuberYT committed:
add getMotorsOkTemperature to DriveSubsystem (again? wtf github)

Hydro111 committed:
Tune LegAnkle PIDs partially

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

# Conflicts:
#	src/main/java/frc/robot/Constants.java

Mothandman committed:
I'm trying to fix my cocdiocdi (sorry Hale)

Mothandman committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

NotAYoutuberYT committed:
remove merge stuff from constants (how did that get there?)

JasperDavidson committed:
Relatively Optimized Reid's Grabber

Still some optimizations to be made, mainly tuning for the PID's

Hydro111 committed:
Adjust code for new wiring

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

# Conflicts:
#	src/main/java/frc/robot/Constants.java
#	src/main/java/frc/robot/RobotContainer.java
#	src/main/java/frc/robot/subsystems/ShuffleboardSubsystem.java

CrystalCanStay committed:
Added seesAprilTag method to PhotonVisionSusystem

CrystalCanStay committed:
Fixed red squiggly

JasperDavidson committed:
Use the code

Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!Nice!!!!!!!!!

Hydro111 committed:
Merge branch 'dev' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into dev

# Conflicts:
#	src/main/java/frc/robot/Constants.java
#	src/main/java/frc/robot/subsystems/LegAnkleSubsystem.java
#	src/main/java/frc/robot/subsystems/ReidPrototypeSubsystem.java

Hydro111 committed:
Add testing changes

Hydro111 committed:
Tune pitch PIDs

May0san committed:
Update Constants.java

May0san reviewed (approved):
10/10 would eat here again. very nice staff and the bathrooms are very clean.

Hydro111 committed:
Uncomment drive train code

Hydro111 committed:
Recomment drive train

Add limit switch logic

JasperDavidson committed:
Should work, current PID's just need to be tuned now

Good luck Hale

Hydro111 committed:
Update version

Hydro111 committed:
Merge branch 'Reid's-Grabber' into HEAD

Hydro111 committed:
Merge branch Reid's-Grabber into branch Testing

Hydro111 committed:
Merge branch 'testing' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into testing

May0san committed:
mleen

did some PID tuning, added an output range on pivot, also made the A button stop applying setpoints to PIDs, so we can test setpoints over long distances

Hydro111 committed:
Fix inverse kinematics (Untested)

Hydro111 committed:
Make small code review changes

Enable arm movement via PID values

Hydro111 committed:
Merge branch 'testing' of https://github.com/amesrobotics/2023-CHARGED-UP-3243 into testing

# Conflicts:
#	src/main/java/frc/robot/Constants.java
#	src/main/java/frc/robot/commands/WristCommand.java
